### PR TITLE
Fix a bug related to InitialBindings

### PR DESCRIPTION
**Pull request by simonvbrae**
Added an undefined check which resolves this issue:
https://github.com/comunica/comunica/issues/1154

### CODE REVIEW

File: lib/sparql.ts [hunk 1]
```diff
@@ -279,12 +279,14 @@ function translateExtend(op: Algebra.Extend): any
 function translateFrom(op: Algebra.From): GroupPattern
 {
     const result = translateOperation(op.input);
-    // project is nested in group object
-    const obj: SelectQuery = result.patterns[0];
-    obj.from = {
-        default: op.default,
-        named: op.named
-    };
+    if (result.patterns[0] !== undefined) {
```
Comment (joachimvh) [inline lib/sparql.ts:282]: Does this issue occur when the first entry of the `patterns` array is `undefined`, or if the `patterns` array is empty, which causes this to result to be `undefined`? If it is the latter, this check should be `result.patterns.length > 0`.

Comment (rubensworks) [reply lib/sparql.ts:282]: Related to this, might make sense to add a unit test for it (or a query in which the problem used to occur).

Comment (simonvbrae) [reply lib/sparql.ts]: > Does this issue occur when the first entry of the `patterns` array is `undefined`, or if the `patterns` array is empty.
I mistyped my check. The issue occurs when the patterns array is undefined I've updated the if condition.

Comment (simonvbrae) [reply lib/sparql.ts]: > might make sense to add a unit test for it (or a query in which the problem used to occur).
To set initialbindings for the test, should I add another function in sparql-test.js?

For completeness, the case which failed was:
```
query = CONSTRUCT FROM <http://www.ontotext.com/retain-bind-position> WHERE { ?s ?p ?o }  LIMIT 100
const quadStream = await this.engine.queryQuads(query, {
  sources: [
        {
          type: 'sparql',
          value: 'https://www.goudatijdmachine.nl/sparql/repositories/gtm',
        },
      ],
  initialBindings: bindingsFactory.fromRecord({}) as unknown as Bindings,
});
```

Comment (joachimvh) [reply lib/sparql.ts]: You need to add the SPARQL query somewhere in the `test/sparql` folder and the expected algebra result in the same location in the other 2 test folders. During unit tests it is checked if the query correctly converts to the algebra and if the algebra correctly converts to the query.

**Pull request closed without merging**